src/ogma/features/nodeGrouping.ts

```diff
@@ -147,12 +157,51 @@ export class NodeGroupingTransformation {
       return;
     }
 
+    // 2 nodes
+    if (subNodes.size === 2) {
+      const radii = subNodes.getAttribute('radius').map(Number);
+      const positions = subNodes.getPosition();
+      const gap = Math.min(...radii);
+      await subNodes.setAttributes([
+        positions[0],
+        {x: positions[0].x + gap + radii[0] + radii[1], y: positions[0].y}
+      ]);
+      return;
+    }
+
     const noEdges = subNodes.getAdjacentEdges({bothExtremities: true}).size === 0;
-    if (noEdges) {
-      await this._runCirclePack(subNodes);
-    } else {
-      await this._runForceLayout(subNodes);
+    if (noEdges) return this._runCirclePack(subNodes);
+    // stars
+    const center = this.isStar(subNodes);
+    if (center) {
+      const satellites = subNodes.filter((n) => n !== center);
+      const positions = this._runCircularLayout({
+        radii: satellites.getAttribute('radius'),
+        cx: center.getAttribute('x'),
+        cy: center.getAttribute('y'),
+        clockwise: false,
+        distanceRatio: 5
+      });
+      const list = center.toList().concat(satellites);
+      await list.setAttributes([center.getPosition(), ...positions]);
+      return;
+    }
+    // Chains: if al nodes have degree 1 or 2, place them in a line
+    const degrees = subNodes.getDegree();
+    if (degrees.every((d) => d > 0 && d <= 2)) {
+      // straighten the chain
+      const sortedNodes = this._ogma.getNodes(this.topologicalSort(subNodes));
+      // we also need to sort the nodes so that they are following the chain
+      await this._ogma.layouts.grid({
+        nodes: sortedNodes,
```

Comment:
in the grid layout, the position of nodes will be defined by their index in the list ?

Reply:
yes